homeassistant/components/klyqa/config_flow.py

```diff
@@ -0,0 +1,307 @@
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_POLLING, CONF_SYNC_ROOMS, DOMAIN, LOGGER
+from .datacoordinator import HAKlyqaAccount
+
+user_step_data_schema = {
+    vol.Required(CONF_USERNAME, default=""): cv.string,
+    vol.Required(CONF_PASSWORD, default=""): cv.string,
+    vol.Required(CONF_SCAN_INTERVAL, default=60): int,
+    vol.Required(
+        CONF_SYNC_ROOMS, default=True, msg="sync r", description="sync R"
+    ): bool,
+    vol.Required(CONF_POLLING, default=True): bool,
+    vol.Required(CONF_HOST, default="https://app-api.prod.qconnex.io"): str,
+}
+
+NoneType = type(None)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a config flow."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.username = ""
+        self.password = ""
+        self.scan_interval = -1
+        self.sync_rooms = False
+        self.polling = False
+        self.host = ""
+        self.klyqa: HAKlyqaAccount | None = None
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            self.username = str(user_input[CONF_USERNAME])
+            self.password = str(user_input[CONF_PASSWORD])
+            self.scan_interval = int(user_input[CONF_SCAN_INTERVAL])
+            self.sync_rooms = user_input[CONF_SYNC_ROOMS]
+            self.polling = user_input[CONF_POLLING]
+            self.host = str(user_input[CONF_HOST])
+            return await self._async_klyqa_login(step_id="user")
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_USERNAME, default=self.config_entry.data[CONF_USERNAME]
+                    ): cv.string,
+                    vol.Required(
+                        CONF_PASSWORD, default=self.config_entry.data[CONF_PASSWORD]
+                    ): cv.string,
+                    vol.Required(
+                        CONF_POLLING, default=self.config_entry.data[CONF_POLLING]
+                    ): bool,
+                    vol.Required(
+                        CONF_SCAN_INTERVAL,
+                        default=self.config_entry.data[CONF_SCAN_INTERVAL],
+                    ): int,
+                    vol.Required(
+                        CONF_SYNC_ROOMS,
+                        default=self.config_entry.data[CONF_SYNC_ROOMS],
+                    ): bool,
+                    vol.Required(
+                        CONF_HOST, default=self.config_entry.data[CONF_HOST]
+                    ): str,
```

Comment:
Same question as with the config flow schema, not sure what a user needs these for?